### ISSUE TITLE
Retext crashes abruptly [ubuntu 18.10, ReText 7.0.1]

### ISSUE THREAD

**Issue by Gepp35** [retext-project/retext#442]
I'm disappointed: I just lost 50 minutes of writing (no so many characters, but around 10 mathjax equations within). I am really disappointed and upset by this situation. Do you plan to get a rescue file or a automatic save routine (every 5 minutes for example)? When I want to fork or a time-point save, I do it consciously but it is when I am absorbed by writing that I will like a safety backup if Retext crashes.

**Comment by mitya57** [comment]
Hi!

I am really sorry that you lost your data because of this bug.

> Do you plan to get a rescue file or a automatic save routine (every 5 minutes for example)?

Yes, ReText provides an auto-save mode. It is the first setting in the Preferences dialog. The documents will be saved every minute.

To help me fix the crash, can you please provide more detail?

- How did you install ReText? Ubuntu 18.10 repository has version 7.0.4, where did you get 7.0.1 from?
- If you installed ReText from Ubuntu repository, do you have any file in `/var/crash/` related to this crash? Can you attach it?
- Were you doing anything special when the crash happened? Maybe you clicked some menu action, or inserted a formula, or something like that?
- Are you able to reproduce this crash? If yes, can you check what is printed to stdout when it happens? (E.g. run `retext` in Terminal and check what is printed there.)

**Comment by Gepp35** [comment]
I tried an apt update:
```
~$ sudo apt install retext
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances       
Lecture des informations d'état... Fait
retext est déjà la version la plus récente (7.0.1-1).
```
it's in french: say that the 7.0.1-1 version is the up-to-date :(

**Comment by mitya57** [comment]
> strange, pretty sure to have installed it from (mate)ubuntu repository (bionic) and update and upgrade every day!

Ah. Bionic is 18.04, not 18.10 like you say in this issue's title. It has an older version, yes. Though I am not aware of any crashes that are present in 7.0.1 but fixed in 7.0.4. There are other less important bugs fixed, but not crashes…

> answer: I was writing text for ~15 minutes only (no special character or format)

That is not enough information to know what caused the crash. So unfortunately I will be able to fix it only if you get it again (see below).

> but I saw that the real time rendering part was slow to render each time I wrote a character.

Do you mean the math formulas rendering? Yes, MathJax is slow, this is not something I can fix in ReText :(

> Launching from console warn: "libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile".

These are harmless warnings from Qt WebKit, you can ignore them.

> Sorry I can't better help you, but now I will update it and launch it in console with automatic save activated (completely missed it!).

Thanks! If it crashes again then please paste what is printed there, either here or in the new issue.

I am closing this issue for now, but will reopen if you get it again.
